### PR TITLE
feat: limit addons dialogue size

### LINKED ISSUE #30: Limit addons dialogue size

### PR DESCRIPTION
**Pull request by Oleksii-Klimov**

### CODE REVIEW

File: aidial_assistant/chain/dialogue.py [hunk 1]
```diff
@@ -1,17 +1,27 @@
-from aidial_assistant.chain.model_client import Message
+from pydantic import BaseModel
+
+from aidial_assistant.model.model_client import Message
+
+
+class Exchange(BaseModel):
```
Comment (adubovik) [inline aidial_assistant/chain/dialogue.py]: How about to call it `DialogueTurn`?

I borrowed it from the [Bison spec](https://cloud.google.com/vertex-ai/docs/generative-ai/learn/models), which specifies max number of *turns*.

File: aidial_assistant/chain/command_chain.py [hunk 2]
```diff
@@ -86,27 +108,39 @@ async def run_chat(self, history: History, callback: ChainCallback):
...
+                all_messages = self._reinforce_json_format(
+                    messages + retries.messages
+                )
+                if model_request_limiter:
+                    await model_request_limiter.verify_limit(all_messages)
+
                 chunk_stream = CumulativeStream(
                     self.model_client.agenerate(
-                        self._reinforce_json_format(messages + retries.messages)
+                        all_messages, **self.model_extra_args  # type: ignore
```
Comment (adubovik) [inline aidial_assistant/chain/command_chain.py:147]: `max_tokens` must be decreased each time we add a new turn to the dialogue by the token size of this turn, right?

File: aidial_assistant/chain/dialogue.py (hunk 1 above)
Comment (Oleksii-Klimov) [reply aidial_assistant/chain/dialogue.py]: Renamed.

File: aidial_assistant/chain/command_chain.py (hunk 2 above)
Comment (Oleksii-Klimov) [reply aidial_assistant/chain/command_chain.py:147]: This parameter limits completions in nested (addon) sessions. It is used for generating commands and replies within these sessions.

**Review by adubovik — APPROVED** [review]

**Pull request merged**